src/main/errorDialog.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Error dialog component
+ *
+ * An Electron BrowserWindow dialog that presents errors to the user.
+ * GPII Application
+ * Copyright 2016 Steven Githens
+ * Copyright 2016-2017 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var fluid   = require("infusion");
+var gpii    = fluid.registerNamespace("gpii");
+var ipcMain = require("electron").ipcMain;
+
+require("./utils.js");
+
+/**
+ * A component that serves as simple interface for communication with the
+ * electron `BrowserWindow` error dialog.
+ */
+fluid.defaults("gpii.app.errorDialog.channel", {
+    gradeNames: ["fluid.component"],
+
+    events: {
+        onErrorDialogCreated: null,
+        onErrorDialogClosed: null,
+        onContentHeightChanged: null
+    },
+
+    listeners: {
+        "onCreate.registerChannel": {
+            funcName: "gpii.app.errorDialog.channel.register",
+            args: ["{that}"]
+        },
+        "onDestroy.deregisterChannel": {
+            funcName: "gpii.app.errorDialog.channel.deregister"
+        }
+    },
+
+    invokers: {
+        update: {
+            funcName: "gpii.app.notifyWindow",
+            args: [
+                "{dialog}.dialog",
+                "onErrorUpdate",
+                "{arguments}.0"
+            ]
+        }
+    }
+});
+
+/**
+ * Register for events from the managed Electron `BrowserWindow` (the renderer process).
+ * @param that {Component} The `gpii.app.errorDialog.channel` instance.
+ */
+gpii.app.errorDialog.channel.register = function (that) {
+    ipcMain.on("onErrorDialogCreated", function () {
+        that.events.onErrorDialogCreated.fire();
+    });
+
+    ipcMain.on("onErrorDialogClosed", function () {
+        that.events.onErrorDialogClosed.fire();
+    });
+
+    ipcMain.on("onErrorDialogHeightChanged", function (event, height) {
+        that.events.onContentHeightChanged.fire(height);
+    });
+};
+
+/**
+ * Removes the IPC listeners needed for the communication with the `BrowserWindow`
+ * when the latter is about to be destroyed.
+ */
+gpii.app.errorDialog.channel.deregister = function () {
+    ipcMain.removeAllListeners("onErrorDialogCreated");
```

Comment:
I think we've got enough of this type of material now that the design would be substantially rationalised with a generalised IoCSS-based brokering system of the kind you were speculating about a couple of months ago. If we find ourselves without higher priority tasks I think we should go for it - let's talk in the week. This is also something which @the-t-in-rtf started to sketch out a couple of years back.